app/frontend/src/components/forms/manage/ApiKey.vue

```diff
@@ -0,0 +1,202 @@
+              </v-btn>
+            </template>
+            <span v-if="showSecret">Hide Secret</span>
+            <span v-else>Show Secret</span>
+          </v-tooltip>
+
+          <BaseCopyToClipboard
+            :disabled="!canReadSecret || !showSecret"
+            class="ml-2"
+            :copyText="secret"
+            snackBarText="Secret copied to clipboard"
+            tooltipText="Copy secret to clipboard"
+          />
+
+          <v-tooltip bottom>
+            <template #activator="{ on, attrs }">
+              <v-btn
+                class="ml-2"
+                color="red"
+                :disabled="!canDeleteKey"
+                icon
+                small
+                v-bind="attrs"
+                v-on="on"
+                @click="showDeleteDialog = true"
+              >
+                <v-icon>delete</v-icon>
+              </v-btn>
+            </template>
+            <span>Delete Key</span>
+          </v-tooltip>
+        </v-col>
+      </v-row>
+    </v-skeleton-loader>
+
+    <!-- Generate/regen confirmation -->
+    <BaseDialog
+      v-model="showConfirmationDialog"
+      type="CONTINUE"
+      @close-dialog="showConfirmationDialog = false"
+      @continue-dialog="createKey"
+    >
+      <template #title>Confirm Key Generation</template>
+      <template #text>
+        <span v-if="!apiKey">
+          Create an API Key for this form? <br />
+          Ensure you follow the Dislaimer on this page.
+        </span>
+        <span v-else>
+          Regenerate the API Key? <br />
+          <strong>Continuing will delete your current API Key access.</strong>.
+        </span>
+      </template>
+      <template #button-text-continue>
+        <span>{{ apiKey ? 'Regenerate' : 'Generate' }} Key</span>
+      </template>
+    </BaseDialog>
+
+    <!-- Delete confirmation -->
+    <BaseDialog
+      v-model="showDeleteDialog"
+      type="CONTINUE"
+      @close-dialog="showDeleteDialog = false"
+      @continue-dialog="deleteKey"
+    >
+      <template #title>Confirm Deletion</template>
+      <template #text> Are you sure you wish to delete your API Key? </template>
+      <template #button-text-continue>
+        <span>Delete</span>
+      </template>
+    </BaseDialog>
+  </div>
+</template>
+
+<script>
+import { mapActions, mapGetters } from 'vuex';
+import { FormPermissions } from '@/utils/constants';
+
+export default {
+  name: 'ApiKey',
+  data() {
+    return {
+      loading: false,
+      showConfirmationDialog: false,
+      showDeleteDialog: false,
+      showSecret: false,
+    };
+  },
+  computed: {
+    ...mapGetters('form', ['apiKey', 'form', 'permissions']),
+    canDeleteKey() {
+      return (
+        this.permissions.includes(FormPermissions.FORM_API_DELETE) &&
+        this.apiKey
+      );
+    },
+    canReadSecret() {
+      return (
+        this.permissions.includes(FormPermissions.FORM_API_READ) && this.apiKey
+      );
+    },
+    canGenerateKey() {
```

Comment:
should we move this canGenerateKey method above the canReadSecret method for ordering?